src/RulesDeploy.js

```diff
@@ -65,7 +74,30 @@ RulesDeploy.prototype = {
         })
       );
     });
-    return Promise.all(promises);
+    return Promise.all(promises).catch(async (err) => {
+      if (err.status === QUOTA_EXCEEDED_STATUS_CODE) {
+        utils.logBullet(
+          clc.bold.yellow(self.type + ":") + " quota exceeded error while uploading rules"
```

Comment:
if the RulesDeploy to TS PR goes in first, remove references to `self` (which can already be done after using arrow functions, but definitely should be updated when moving to TS)